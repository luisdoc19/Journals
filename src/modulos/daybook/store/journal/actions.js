import journalApi from "@/api/journalApi";

export const loadEntries = async ({ commit }) => {
  const { data } = await journalApi.get("/entries.json");
  if (!data) {
    commit("setEntries", []);
    return;
  }

  const entries = [];
  for (let id of Object.keys(data)) {
    entries.push({
      id,
      ...data[id],
    });
  }
  commit("setEntries", entries);
};
export const updateEntry = async ({ commit }, entry) => {
  const { id, date, picture, text } = entry;
  const updatedEntry = {
    date,
    picture,
    text,
  };
  const { data } = await journalApi.put(`/entries/${id}.json`, updatedEntry);
  commit("updateEntry", { id, ...data });
};
export const createEntry = async ({ commit }, entry) => {
  const { date, text, picture } = entry;
  const newEntry = {
    date,
    text,
    picture,
  };
  const { data } = await journalApi.post(`/entries.json`, newEntry);
  newEntry.id = data.name;
  commit("addEntry", newEntry);
  return newEntry.id;
};
export const deleteEntry = async ({ commit }, id) => {
  try {
    await journalApi.delete(`/entries/${id}.json`);
    commit("deleteEntry", id);
    return true;
  } catch (error) {
    console.log(error);
    return false;
  }
};
