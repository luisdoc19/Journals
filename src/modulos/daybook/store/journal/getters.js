export const getEntriesByTerm =
  (state) =>
  (term = "") => {
    if (term.length === 0) return state.entries;
    return state.entries.filter((entry) =>
      entry.text.toLowerCase().includes(term.toLowerCase())
    );
  };

export const getEntryById = (state) => (val) => {
  const entry = state.entries.find((entry) => entry.id === val);
  if (!entry) return;
  return { ...entry };
};
