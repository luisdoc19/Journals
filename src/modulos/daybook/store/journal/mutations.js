export const setEntries = (state, entries) => {
  state.entries = [...state.entries, ...entries];
  state.isLoading = false;
};
export const updateEntry = (state, entry) => {
  const index = state.entries.findIndex((e) => e.id === entry.id);
  state.entries[index] = entry;
};
export const addEntry = (state, entry) => {
  state.entries = [...state.entries, entry];
};
export const deleteEntry = (state, id) => {
  state.entries = state.entries.filter((e) => e.id !== id);
};

export const clearEntries = (state) => {
  state.entries = [];
};
