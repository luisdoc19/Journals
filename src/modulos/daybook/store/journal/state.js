export default () => ({
  isLoading: true,
  entries: [
    {
      id: new Date().getTime(),
      date: new Date().toDateString(),
      text: "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Eligendi rerum quos veniam debitis iure quae dignissimos ex nobis aspernatur in numquam id cumque inventore sunt, necessitatibus odit delectus cupiditate maxime!",
      picture: null,
    },
    {
      id: new Date().getTime() + 1000,
      date: new Date().toDateString(),
      text: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Saepe neque minus quam sequi. Nesciunt vero aliquid non quaerat doloremque nisi fugit aperiam velit blanditiis atque? Aut expedita repellendus labore quo?",
      picture: null,
    },
    {
      id: new Date().getTime() + 2000,
      date: new Date().toDateString(),
      text: "Lorem, ipsum dolor sit amet consectetur adipisicing elit. In, est dignissimos laudantium harum, repellendus perferendis at labore, nihil maxime quo sint perspiciatis ab ullam deleniti dolorem illum asperiores voluptatem veniam.",
      picture: null,
    },
  ],
});
