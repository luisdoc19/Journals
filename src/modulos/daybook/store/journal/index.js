import state from "./state";
import * as getters from "./getters";
import * as mutations from "./mutations";
import * as actions from "./actions";

const myCustomModule = {
  namespaced: true,
  state,
  getters,
  mutations,
  actions,
};

export default myCustomModule;
