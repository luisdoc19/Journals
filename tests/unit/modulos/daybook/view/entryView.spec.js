import { createStore } from "vuex";
import { shallowMount } from "@vue/test-utils";
import EntryViewVue from "@/modulos/daybook/views/EntryView.vue";
import journal from "@/modulos/daybook/store/journal";
import { jornalState } from "../../../mock-data/test-jornal";
import Swal from "sweetalert2";

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });

jest.mock("sweetalert2", () => ({
  fire: jest.fn(),
  showLoading: jest.fn(),
  close: jest.fn(),
}));

describe("EntryView", () => {
  const store = createVuexStore(jornalState);
  store.dispatch = jest.fn();
  const mockRouter = {
    push: jest.fn(),
  };

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryViewVue, {
      props: {
        id: "-Nb7MoPaZPjx5_ULLjgb",
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });
  it("debe de sacar al usuario porque el id no existe", () => {
    const wrapper = shallowMount(EntryViewVue, {
      props: {
        id: "Este id no existe en el Store",
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
    console.log(wrapper.html());
    expect(mockRouter.push).toHaveBeenCalled();
  });

  it("debe de mostrar la entrada correcta", () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(mockRouter.push).not.toHaveBeenCalled();
  });

  it("debe de borrar la entrada y salir", (done) => {
    Swal.fire.mockReturnValueOnce(Promise.resolve({ isConfirmed: true }));
    wrapper.find(".btn-danger").trigger("click");
    expect(Swal.fire).toHaveBeenCalled();
    setTimeout(() => {
      expect(store.dispatch).toHaveBeenCalledWith(
        "journal/deleteEntry",
        "-Nb7MoPaZPjx5_ULLjgb"
      );
      expect(mockRouter.push).toHaveBeenCalled();
      done();
    }, 1);
  });
});
