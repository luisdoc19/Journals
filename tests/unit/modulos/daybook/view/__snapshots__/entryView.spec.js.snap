// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryView debe de mostrar la entrada correcta 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">05</span><span class="mx-1 fs-3">Aug</span><span class="mx-2 fs-3 fw-ligth">2023</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Qué sucedió hoy?"></textarea></div>
<img src="https://res.cloudinary.com/djrggifvj/image/upload/v1691284081/pvwxc3bumtvjmnmsfdcs.png" alt="entry-picture" class="img-thumbnail">
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
